### PR TITLE
Clean-up SimpleListeners and fix invalid XML header

### PR DESCRIPTION
**Pull request by premun**
As mentioned here: https://github.com/dotnet/xharness/issues/91#issuecomment-619859563, we include some garbage in the results XML

### CODE REVIEW

File: tests/Microsoft.DotNet.XHarness.iOS.Tests/AppRunnerTests.cs [hunk 1]
```diff
@@ -274,7 +274,6 @@ public async Task RunOnSimulatorSuccessfullyTest()
             _listener.Verify(x => x.Initialize(), Times.AtLeastOnce);
             _listener.Verify(x => x.StartAsync(), Times.AtLeastOnce);
             // _listener.Verify(x => x.Cancel(), Times.AtLeastOnce); // TODO: https://github.com/dotnet/xharness/issues/73
-            _listener.Verify(x => x.Dispose(), Times.AtLeastOnce);
```
Comment (mandel-macaque) [inline tests/Microsoft.DotNet.XHarness.iOS.Tests/AppRunnerTests.cs:277]: Why was this removed? should the app runner be disposing the listener?

File: tests/Microsoft.DotNet.XHarness.iOS.Tests/AppRunnerTests.cs [hunk 2]
```diff
@@ -402,7 +401,6 @@ public async Task RunOnDeviceSuccessfullyTest(bool useTunnel)
...
```
Comment (mandel-macaque) [inline tests/Microsoft.DotNet.XHarness.iOS.Tests/AppRunnerTests.cs:405]: Same, why did we remove this?

File: tests/Microsoft.DotNet.XHarness.iOS.Tests/AppRunnerTests.cs (hunk 1 above)
Comment (premun) [reply tests/Microsoft.DotNet.XHarness.iOS.Tests/AppRunnerTests.cs:277]: A mistake - bad revert, will fix
